tests/src/python/test_provider_ogr_gpkg.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+"""QGIS Unit tests for the OGR/GPKG provider.
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Even Rouault'
+__date__ = '2016-04-21'
+__copyright__ = 'Copyright 2016, Even Rouault'
+# This will get replaced with a git SHA1 when you do a git archive
+__revision__ = '$Format:%H$'
+
+import qgis  # NOQA
+
+import os
+import tempfile
+import shutil
+import glob
+from osgeo import gdal, ogr
+
+from qgis.core import QgsVectorLayer, QgsFeature, QgsGeometry, QgsFeatureRequest
+from qgis.testing import start_app, unittest
+from utilities import unitTestDataPath
+
+start_app()
+
+
+def GDAL_COMPUTE_VERSION(maj, min, rev):
+    return ((maj) * 1000000 + (min) * 10000 + (rev) * 100)
+
+
+class TestPyQgsOGRProviderGpkg(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """Run before all tests"""
+        # Create test layer
+        cls.basetestpath = tempfile.mkdtemp()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Run after all tests"""
+        shutil.rmtree(cls.basetestpath, True)
+
+    @unittest.expectedFailure(int(gdal.VersionInfo('VERSION_NUM')) < GDAL_COMPUTE_VERSION(1, 11, 0))
```

Comment:
I didn't want to put constraints on the minimum GDAL version since it seems that the code base actually supports older one (I accidentaly compiled it against the GDAL 1.10 from one of my systems I think). The entry in the menubar to create a new GPKG layer is not present if GDAL < 1.11.
Are you suggesting to bump the compile-time requirement to GDAL 1.11 or just let the test fail (I'm not sure all packagers run the test suite) ? I'm fine with whatever the project decides.